### PR TITLE
39 add custom functions to jmespath

### LINKED ISSUE #39: Add `upper`, `lower` and `capitalize` custom functions to JMESPath

### PR DESCRIPTION
**Pull request by spicy-sauce**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,6 +122,9 @@ DataYoga supports both SQL and [JMESPath](https://jmespath.org/) expressions. JM
...
+| ------------ | ----------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------- |
+| `capitalize` | Capitalizes all the words in the string                                 | Input: `{name: "john doe"}` <br /> Expression: `capitalize(name)` <br /> Output: `John Doe`                     |
+| `concat`     | Concatenates an array of variables or literals                          | Input: `{fname: "john", lname: "doe"}` <br /> Expression: `concat([fname,' ',lname])` <br /> Output: `john doe` | This is equivalent to the more verbose built-in expression: `' '.join([fname,lname])`. |
+| `lower`      | Converts all uppercase characters in a string into lowercase characters | Input: `{fname: "John"}` <br /> Expression: `upper(fname)` <br /> Output: `john`                                |
```
Comment (zalmane) [inline README.md]: example should be 'lower' (instead of 'upper')

File: tests/test_jmespath_expression.py [hunk 2]
```diff
@@ -1,7 +1,26 @@
...
-def test_sql_expression():
-    expression = JMESPathExpression()
-    expression.compile("concat([fname,' ',mname,' ',lname])")
-    assert expression.search({"fname": "john",  "mname": "george", "lname": "smith"}) == "john george smith"
+CONCAT_EXPRESSION = "concat([fname,' ',mname,' ',lname])"
+TEST_JSON = {"fname": "JohN",  "mname": "GeorgE", "lname": "SmitH"}
+
+
+def test_jmespath_concat_expression():
+    expression.compile(CONCAT_EXPRESSION)
```
Comment (zalmane) [inline tests/test_jmespath_expression.py:10]: any reason to have this as a global variable? is it reused?

File: tests/test_jmespath_expression.py [hunk 3]
```diff
@@ -1,7 +1,26 @@
...
+    assert expression.search(TEST_JSON) == "JohN GeorgE SmitH"
+
+
+def test_jmespath_capitalize_expression():
+    expression.compile(f"capitalize({CONCAT_EXPRESSION})")
```
Comment (zalmane) [inline tests/test_jmespath_expression.py]: any reason to mix the concat in the test? Ideally the test should be focused on one thing, so maybe remove the concat from the upper/lower tests?

File: README.md (hunk 1 above)
Comment (spicy-sauce) [reply README.md]: thanks, my bad.

File: tests/test_jmespath_expression.py (hunk 3 above)
Comment (spicy-sauce) [reply tests/test_jmespath_expression.py]: fixed

**Review by zalmane — APPROVED** [review]

**Pull request merged**